### ISSUE TITLE
Add top-k classification accuracy metrics

### ISSUE THREAD

**Issue by ramananbalakrishnan** [fchollet/keras#3986]
### Feature proposal

Adding a metric to track top-k classification accuracy.
### Details

Seeing how metrics like top-5 error rates are increasing in popularity, I was thinking that a metric to track top-k classification performance might be useful. Useful when the `argmax()` prediction might not match, but subsequent predictions might.

Given that TensorFlow already has a similar metric [`tf.nn.in_top_k()`](https://www.tensorflow.org/versions/r0.11/api_docs/python/nn.html#in_top_k), it should be possible to include this into keras as well.

I will try to write a simple implementation for this feature and send in a pull request.

---

Please make sure that the boxes below are checked before you submit your issue. Thank you!
- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
  pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps
- [x] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
  pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps
- [x] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).
